tests/testthat/test-transitivity.R

```diff
@@ -27,3 +27,21 @@ test_that("no integer overflow", {
   mtr <- min(transitivity(g, type = "local"), na.rm = TRUE)
   expect_true(mtr > 0)
 })
+
+# Check that transitivity() produces named vectors, see #943
+# The four tests below check four existing code paths
+test_that("local transitivity produces named vectors", {
+  g <- make_graph(~ a-b-c-a-d)
+  E(g)$weight <- 1:4
+  t1 <- transitivity(g, type = "local")
+  t2 <- transitivity(g, type = "barrat")
+
+  vs <- c("a", "c")
+  t3 <- transitivity(g, type = "local", vids = vs)
+  t4 <- transitivity(g, type = "barrat", vids = vs)
+
+  expect_equal(names(t1), V(g)$name)
+  expect_equal(names(t2), V(g)$name)
+  expect_equal(names(t3), vs)
+  expect_equal(names(t4), vs)
```

Comment:
I have a slight preference for

```suggestion
  t1 <- transitivity(g, type = "local")
  expect_equal(names(t1), V(g)$name)
  
  t2 <- transitivity(g, type = "barrat")
  expect_equal(names(t2), V(g)$name)

  vs <- c("a", "c")
  t3 <- transitivity(g, type = "local", vids = vs)
  expect_equal(names(t3), vs)
  
  t4 <- transitivity(g, type = "barrat", vids = vs)
  expect_equal(names(t4), vs)
```

because this keeps input and output closer together. No need to act here.